tools/about_paint.py

```diff
@@ -0,0 +1,117 @@
+from collections import Counter
+import copy
+from fontTools.colorLib.builder import LayerListBuilder
+from fontTools.ttLib import TTFont
+from fontTools.ttLib.tables import otTables as ot
+from collections import defaultdict
+from itertools import groupby
+import logging
+from math import acos, asin, atan, degrees
+import os
+import sys
+
+def traverse_paint(colr, callback):
+    """Traverse and optionally rewrite Paint in a COLRv1 graph."""
+    if not callable(callback):
+        raise TypeError("callback must be callable")
+
+    visited = set()
+    frontier = []
+    # Glyph graphs start from either base glyphs or the layerlist
+    for base_glyph in colr.BaseGlyphList.BaseGlyphPaintRecord:
+        frontier.append((base_glyph, base_glyph.Paint))
+    for paint in colr.LayerList.Paint:
+        frontier.append((colr.LayerList.Paint, paint))
+    while frontier:
+        container, paint = frontier.pop()
+        key = (id(container), id(paint))
+        if key in visited:
+            continue
+        visited.add(key)
+
+        new_paint = callback(paint)
+        if new_paint is not None and new_paint is not paint:
+            # replace existing paint
+            did_replace = False
+            if isinstance(container, list):
+                idx = container.index(paint)
+                container[idx] = new_paint
+                did_replace = True
+            else:
+                for conv in container.getConverters():
+                    value = getattr(container, conv.name, None)
+                    if value is paint:
+                        setattr(container, new_paint)
+                        did_replace = True
+            if not did_replace:
+                raise ValueError("Unable to replace, no current field found")
+            paint = new_paint
+        frontier.extend((paint, c) for c in reversed(paint.getChildren(colr)))
+
+def f2dot14_safe(*values):
+    return all(value >= -2.0 and value < 2.0 for value in values)
+
+def _angle(arcfn, v):
+    return round(degrees(arcfn(v)), 4)
+
+
+def _common_angle(a, b, c, d):
+    return (
+        all(-1 <= v <= 1 for v in (a, b, c, d))
+        and len({
+            _angle(acos, a),
+            _angle(asin, b),
+            _angle(lambda v: -asin(v), c),
+            _angle(acos, d)
+        }) == 1
+    )
+
+colr_builder = LayerListBuilder()
+
+def _tuple(paint):
+    return list_builder._paint_suble(paint)
```

Comment:
typo?

```suggestion
    return list_builder._paint_tuple(paint)
```

Reply:
`list_builder` name is undefined. I think the whole `_tuple(paint)` helper function is not used

Reply:
Quite right, copy/paste error

Reply:
Fixed